src/checker/zx/ZXChecker.cpp

```diff
@@ -0,0 +1,147 @@
+#include "checker/zx/ZXChecker.hpp"
+
+#include "Definitions.hpp"
+#include "QuantumComputation.hpp"
+#include "Simplify.hpp"
+#include "ZXDiagram.hpp"
+#include "dd/Definitions.hpp"
+#include "zx/FunctionalityConstruction.hpp"
+
+#include <chrono>
+#include <cstddef>
+#include <iostream>
+#include <numeric>
+#include <optional>
+#include <unordered_set>
+
+namespace ec {
+    ZXEquivalenceChecker::ZXEquivalenceChecker(const qc::QuantumComputation& qc1, const qc::QuantumComputation& qc2, Configuration configuration) noexcept:
+        EquivalenceChecker(qc1, qc2, std::move(configuration)), miter(zx::FunctionalityConstruction::buildFunctionality(&qc1)), tolerance(configuration.functionality.traceThreshold) {
+        zx::ZXDiagram dPrime = zx::FunctionalityConstruction::buildFunctionality(&qc2);
+
+        if (qc1.getNancillae() != 0 || qc2.getNancillae() != 0)
+            ancilla = true;
+
+        auto p1 = invertPermutations(qc1);
+        auto p2 = invertPermutations(qc2);
+
+        // fix ancillaries to |0>
+        for (auto anc = static_cast<dd::Qubit>(qc1.getNqubits() - 1); anc >= qc1.getNqubitsWithoutAncillae(); --anc) {
+            miter.makeAncilla(anc, p1.at(anc));
+            dPrime.makeAncilla(anc, p2.at(anc));
+        }
+        miter.invert();
+        miter.concat(dPrime);
+    }
+
+    EquivalenceCriterion ZXEquivalenceChecker::run() {
+        const auto start = std::chrono::steady_clock::now();
+
+        zx::fullReduceApproximate(miter, tolerance);
+
+        bool equivalent = true;
+
+        if (miter.getNEdges() == miter.getNQubits()) {
+            const qc::Permutation  p1 = invert(invertPermutations(qc1));
+            const qc::Permutation& p2 = invert(invertPermutations(qc2));
```

Comment:
```suggestion
            const auto&  p1 = invert(invertPermutations(qc1));
            const auto& p2 = invert(invertPermutations(qc2));
```
not the missing ampersand on the top line